plutus-core/plutus-core/src/Crypto.hs

```diff
@@ -17,3 +28,25 @@ verifySignature pubKey msg sig =
             <$> publicKey pubKey
             <*> pure msg
             <*> signature sig
+
+verifySECP256k1Signature
+  :: BS.ByteString -- ^ Public key
+  -> BS.ByteString -- ^ Signature
+  -> BS.ByteString -- ^ Message hash
+  -> Emitter (EvaluationResult Bool)
+verifySECP256k1Signature pk sig msg =
+  case DSIGN.rawDeserialiseVerKeyDSIGN @SECP256k1DSIGN pk of
+    Nothing -> do
+      emitM "SECP256k1: Given invalid signing key."
+      pure EvaluationFailure
+    Just pk' -> case DSIGN.rawDeserialiseSigDSIGN @SECP256k1DSIGN sig of
+      Nothing -> do
+        emitM "SECP256k1: Given invalid signature."
+        pure EvaluationFailure
+      Just sig' -> case SECP.msg msg of
+        Nothing -> do
+          emitM "SECP256k1: Given invalid message hash."
+          pure EvaluationFailure
+        Just msg' -> case DSIGN.verifyDSIGN () pk' msg' sig' of
```

Comment:
Oh wait!  `verifySECP256k1Signature` takes its arguments in the order pk/sig/msg, but `DSIGN.verifyDSIGN` takes them in the order pk'/msg'/sig': see also the source of the instance [here](https://github.com/input-output-hk/cardano-base/blob/master/cardano-crypto-class/src/Cardano/Crypto/DSIGN/Ed25519.hs#L145).  Earlier I thought both of these took their arguments in the same order.  Would it be reasonable to change the order of the arguments in the main function?  That would make it consistent with the order expected by `DSIGN` (and also with the existing `verifySignature` function), which would be helpful if we ever implement other signature schemes covered by `DSIGN`.